src/box/applier.cc

```diff
@@ -2553,9 +2558,10 @@ applier_f(va_list ap)
 	bool was_anon = true;
 
 	/* Re-connect loop */
-	while (!fiber_is_cancelled()) {
+	while (true) {
 		FiberGCChecker gc_check;
 		try {
+			fiber_testcancel();
```

Comment:
Do we really need this extra check? AFAIU `applier_connect()` already checks if the fiber is cancelled implicitly, via coio.

Reply:
Discussed f2f and decided to leave as is because it is a good style to have fiber cancel test right after sleep. Also added a comment:

```diff
--- a/src/box/applier.cc
+++ b/src/box/applier.cc
@@ -2562,6 +2562,10 @@ applier_f(va_list ap)
        while (true) {
                FiberGCChecker gc_check;
                try {
+                       /*
+                        * Test cancel after reconnection sleep at the end
+                        * of the loop here to use catch for FiberIsCancelled.
+                        */
                        fiber_testcancel();
                        applier_connect(applier);
                        if (tt_uuid_is_nil(&REPLICASET_UUID)) {
```